Mathlib/Combinatorics/SimpleGraph/Subgraph.lean

```diff
@@ -774,21 +774,19 @@ theorem degree_eq_one_iff_unique_adj {G' : Subgraph G} {v : V} [Fintype (G'.neig
   rw [← finset_card_neighborSet_eq_degree, Finset.card_eq_one, Finset.singleton_iff_unique_mem]
   simp only [Set.mem_toFinset, mem_neighborSet]
 
+lemma neighborSet_eq_of_equiv {v : V} {H : Subgraph G}
```

Comment:
Perhaps a combinatorialist should check if this lemma makes sense mathematically. The golfing looks fine to me.

Reply:
@YaelDillies what do you think?

Reply:
It's a bit weird to state it using an equivalence `G.neighborSet v ≃ H.neighborSet v`, but I guess the existing lemma already does that, so the golf looks 👍